tests/ctest_main.cpp

```
@@ -28,10 +28,29 @@
 
 #include "core.hpp"
 
+[[noreturn]] static bool assertionCallback(const char * message) noexcept {
+  throw std::runtime_error(message);
```

Cppcheck (2.21.0)

[error] Unhandled exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)